main.tf

```diff
@@ -102,6 +102,10 @@ resource "azurerm_kubernetes_cluster" "main" {
       log_analytics_workspace_id = var.enable_log_analytics_workspace ? azurerm_log_analytics_workspace.main[0].id : null
     }
 
+    open_service_mesh {
+      enabled = var.enable_open_service_mesh
+    }
+    
```

Comment:
Looking at the documentation this is not a block:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#open_service_mesh_enabled

It should be a oneliner:

`open_service_mesh_enabled = var.enable_open_service_mesh`

Reply:
@zioproto you're right. For now this module works with provider 2.x so this patch is correct, but we're close to merge a pr that upgrade provider to 3.x so I'm not sure whether we should continue work on 2.x. My thought is we upgrade to 3.x first then let's see what we can do for this pr.